### ISSUE TITLE
Catch subprocess.CalledProcessError

### ISSUE THREAD

**Issue by karlch** [jeanggi90/RawPrev#2]
We should probably catch the `CalledProcessError` and fail to load the image gracefully. This could probably be done by just returning an empty pixmap and let vimiv do the rest.

**Comment by jcjgraf** [comment]
In b163ab5022d16bd734a136a6269b4f689c61fb2b I have switched to `QProcess`. Though, I still need to add some error catching. My naive approach using `QObject().connect(process, process.errorOccured, lambda: print("Error"))` did not seem to be quite right. I gotta research that...

**Comment by karlch** [comment]
Nice, I am positively surprised that using `QProcess` with `QImageReader` fixes the rotation problem.

How about doing something simple like this for the error handling:
```
...
    process.waitForFinished()

    if process.exitStatus() != QProcess.NormalExit or process.exitCode() != 0:
        stderr = process.readAllStandardError()
        raise ValueError(f"Error calling dcraw: '{stderr.data().decode()}'")

    handler = QImageReader(process, "jpeg".encode())
...
```

**Comment by jcjgraf** [comment]
The switch to `QProcess` did not solve the problem itself, but it allowed to to pass the stdin from `dcraw` to `QImageReader` more easily. Since `dcraw` attaches some limited exif information to the extracted jpeg, including the rotation, this way I can make use of the `QImageReader.setAutoTransform()` functionality and rotate the image as need.
And as a very positive side-effect I have noted that `QProcess` does not block the GUI, as `subprocess` did, and hence user experience is much better.

Thanks for you code snipped. This error check is exactly what I was looking for. Unfortunately, `readAllStandardError()` is always an empty string. Changing the `ProcessChannelMode` did not change anything. However, I thinks that is alright.

More of a concern is that simply returning an empty `QPixmap` (as you have suggested) breaks vimiv:

```
Error calling dcraw: ''
[15:09:53] ERROR    <vimiv>              Uncaught exception! Exiting gracefully and printing stack...
Traceback (most recent call last):
  File "vimiv-qt/vimiv/gui/image.py", line 117, in _load_pixmap
    self._update_scene(item, item.boundingRect(), reload_only)
  File "vimiv-qt/vimiv/gui/image.py", line 143, in _update_scene
    self.scale(self._scale if reload_only else ImageScale.Overzoom)  # type: ignore
  File "vimiv-qt/vimiv/gui/image.py", line 250, in scale
    self._scale_to_fit(
  File "vimiv-qt/vimiv/gui/image.py", line 279, in _scale_to_fit
    xratio = self.viewport().width() / width if width is not None else INF
ZeroDivisionError: float division by zero
```
I am actually not sure how vimiv should react when an external reader fails to read an image which it should be able to read.

**Comment by karlch** [comment]
This makes a lot of sense, thanks for the info!

I still had a small bug in the thumbnail manager, but raising a `ValueError` as in the example above should work just fine.

I have had non-empty strings in `readAllStandardError`, e.g. when just appending a useless flag to `dcraw` I get: `Unknown option "--".`. Not sure in which occasions the error message is actually empty though, but I guess this is up to `dcraw` and we cannot really do much as you say.

**Comment by jcjgraf** [comment]
Oh I see. Now it indeed does work with `raise ValueError()`.

I have only tried to run it without having `dcraw` installed and in that case stderr was empty.

Anyways, thanks for you help!

Edit: fixed in 12a5a34247801d38c1b445998f5c81f403745852
